### PR TITLE
feat(kms): add service key dashboard

### PR DESCRIPTION
**Pull request by rjamet-ovh**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/key-management-service-B2`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-13765
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

Add okms service key dashboard page

### CODE REVIEW

File: packages/manager/apps/key-management-service/src/types/okmsServiceKey.type.ts [hunk 1]
```diff
@@ -1,18 +1,144 @@
...
+//   id: string;
+//   createdAt: string;
+//   curve: string | null;
+//   keys: [];
+//   name: string;
+//   operations: string[];
+//   size: number | null;
+//   state: string;
+//   type: string;
+// };
```
Comment (darsene) [inline packages/manager/apps/key-management-service/src/types/okmsServiceKey.type.ts:13]: do you need to keep that comment ?

File: packages/manager/apps/key-management-service/src/types/okmsServiceKey.type.ts [hunk 2]
```diff
@@ -1,18 +1,144 @@
...
+/*
+KEY STATE
+*/
+export type OkmsServiceKeyState =
+  | 'ACTIVE'
+  | 'COMPROMISED'
+  | 'DEACTIVATED'
+  | 'DESTROYED'
+  | 'DESTROYED_COMPROMISED'
+  | 'PRE_ACTIVE';
```
Comment (darsene) [inline packages/manager/apps/key-management-service/src/types/okmsServiceKey.type.ts:50]: cleaner to define an `enum`

File: packages/manager/apps/key-management-service/src/types/okmsServiceKey.type.ts [hunk 3]
```diff
@@ -1,18 +1,144 @@
...
+KEY OPERATIONS
+*/
+
+/* ENCRYPT / DECRYPT */
+export type OkmsServiceKeyOperationEncrypt = 'encrypt';
+export type OkmsServiceKeyOperationDecrypt = 'decrypt';
+export type OkmsServiceKeyOperationEncryptDecrypt = [
+  OkmsServiceKeyOperationEncrypt,
+  OkmsServiceKeyOperationDecrypt,
+];
```
Comment (darsene) [inline packages/manager/apps/key-management-service/src/types/okmsServiceKey.type.ts:62]: transform into `enum`

File: packages/manager/apps/key-management-service/src/types/okmsServiceKey.type.ts [hunk 4]
```diff
@@ -1,18 +1,144 @@
...
+  OkmsServiceKeyOperationVerify,
+];
+
+/* WRAP / UNWRAP */
+export type OkmsServiceKeyOperationWrap = 'wrapKey';
+export type OkmsServiceKeyOperationUnwrap = 'unwrapKey';
+export type OkmsServiceKeyOperationWrapUnwrap = [
+  OkmsServiceKeyOperationWrap,
+  OkmsServiceKeyOperationUnwrap,
+];
```
Comment (darsene) [inline packages/manager/apps/key-management-service/src/types/okmsServiceKey.type.ts:78]: transform into `enum`

File: packages/manager/apps/key-management-service/src/utils/serviceKey/validateServiceKeyName.spec.ts
Comment (darsene) [inline packages/manager/apps/key-management-service/src/utils/serviceKey/validateServiceKeyName.spec.ts:1]: can be simplified with https://vitest.dev/api/#test-each

File: packages/manager/apps/key-management-service/src/components/dashboard/tileValue/tileValue.component.tsx
Comment (darsene) [inline packages/manager/apps/key-management-service/src/components/dashboard/tileValue/tileValue.component.tsx:1]: should be used in all pages then (KMS dashoard, ...)

**Review by vovh — APPROVED** [review]

File: packages/manager/apps/key-management-service/src/pages/dashboard/generalInformations/GeneralInformations.tsx [hunk 5]
```diff
@@ -26,7 +26,7 @@ function GeneralInformationsTab() {
   }
 
   return (
-    <div className="grid xs:grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6">
+    <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 md:gap-6">
```
Comment (darsene) [inline packages/manager/apps/key-management-service/src/pages/dashboard/generalInformations/GeneralInformations.tsx:29]: what about mobile view ?

File: packages/manager/apps/key-management-service/src/pages/dashboard/index.tsx [hunk 6]
```diff
@@ -40,7 +40,7 @@ export default function DashboardPage() {
   ];
 
   return (
-    <>
+    <div className="m-10">
```
Comment (darsene) [inline packages/manager/apps/key-management-service/src/pages/dashboard/index.tsx:43]: Should use `<DashboardLayout />` rather

**Comment by darsene** [comment]
@rjamet-ovh : add some unit tests  to your custom components

**Review by vovh — CHANGES_REQUESTED** [review]
There is some sonar issues to fix

File: packages/manager/apps/key-management-service/src/pages/dashboard/generalInformations/GeneralInformations.tsx (hunk 5 above)
Comment (rjamet-ovh) [reply packages/manager/apps/key-management-service/src/pages/dashboard/generalInformations/GeneralInformations.tsx:29]: classes without prefix are applied on mobile, so it's a `grid-cols-1` setting.

File: packages/manager/apps/key-management-service/src/components/serviceKey/modals/ServiceKeyEditNameModal.component.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+      >
+        {tCommon('key_management_service_cancel')}
+      </OsdsButton>
+      <OsdsButton
+        disabled={
+          !!serviceKeyNameError ||
+          isPending ||
+          serviceKeyName === name ||
+          undefined
+        }
```
[the remark this answers is not in the document]
Comment (rjamet-ovh) [reply packages/manager/apps/key-management-service/src/components/serviceKey/modals/ServiceKeyEditNameModal.component.tsx:120]: without the `undefined` condition, the button is always disabled

File: packages/manager/apps/key-management-service/src/components/dashboard/tileValue/tileValue.component.tsx
Comment (rjamet-ovh) [reply packages/manager/apps/key-management-service/src/components/dashboard/tileValue/tileValue.component.tsx:1]: will do in a another refactoring as kms dashboard are outside the scope of this PR

**Review by vovh — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**